dask_expr/io/parquet.py

```diff
@@ -265,3 +283,82 @@ def _filtered_task(self, index: int):
         if self._series:
             return (operator.getitem, tsk, self.columns[0])
         return tsk
+
+
+#
+# Filtering/Predicate utilities
+#
+
+
+class Or(frozenset):
+    """Helper class for filter disjunctions"""
+
+    def to_list_tuple(self):
+        # NDF "or" is List[List[Tuple]]
+        def _maybe_list(val):
+            if isinstance(val, tuple) and val and isinstance(val[0], (tuple, list)):
+                return list(val)
+            return [val]
+
+        return [
+            _maybe_list(val.to_list_tuple())
+            if hasattr(val, "to_list_tuple")
+            else _maybe_list(val)
+            for val in self
+        ]
+
+
+class And(frozenset):
+    """Helper class for filter conjunctions"""
+
+    def to_list_tuple(self):
+        # NDF "and" is List[Tuple]
+        return tuple(
+            val.to_list_tuple() if hasattr(val, "to_list_tuple") else val
+            for val in self
+        )
```

Comment:
These classes confuse me a little.  They're very similarly named to AND and OR above, and do very little I think.  My programming brain wants to find some way to remove them.  

However, I'm also happy to avoid that work and I'll care much less about them if we're able to isolate these classes to the `to_dnf` function (defined) and so make them very private.  Is that doable?

Reply:
Yes, I was thinking something similar. These classes are a very useful way to make the DNF logic easier to think through, but but it seems critical that these "helper" classes don't get confused with proper expressions.

When I take another pass over this, I'll try to refactor things a bit to make the distinction clearer.

Reply:
If we move them within the `to_dnf` function then I'll stop caring entirely about them :)

Reply:
It was difficult to move everything into a function, but I was able to isolate everything within a general `_DNF` utility class.